src/main/java/com/avanza/gs/test/EmbeddedSpace.java

```diff
@@ -15,58 +15,17 @@
  */
 package com.avanza.gs.test;
 
-import java.util.UUID;
-
-import org.junit.rules.TestRule;
-import org.junit.runner.Description;
-import org.junit.runners.model.Statement;
-import org.openspaces.core.GigaSpace;
-import org.openspaces.core.GigaSpaceConfigurer;
-import org.openspaces.core.space.UrlSpaceConfigurer;
-
-public class EmbeddedSpace implements TestRule {
-
-	private final UrlSpaceConfigurer urlSpaceConfigurer;
-	private final GigaSpace gigaSpace;
+/**
+ * @deprecated use {@link com.avanza.gs.test.junit4.EmbeddedSpace} for JUnit 4

```

Comment:
I think the idea of allowing an implementing app to choose whether to use junit4 or junit5 is really nice. So that's great.

I'm looking at the 375 imports of this particular class at Avanza (`import com.avanza.gs.test.EmbeddedSpace;`). This would mean that those 375 places would get deprecation warnings once they upgrade this lib. At some point we would likely want to suggest junit5 in the deprecation warning instead of junit4. And once we do that, we would probably also want to have good support in Astrix and other internal libraries for junit 5. But until that, apps would probably use a combination of junit4 and 5.

How would you feel about this MR only adding support for junit5 for apps that want to use that, while keeping the old classes in their old package name without deprecation warnings? And later once we start suggesting junit5 instead of junit4, then we add deprecation warnings here? The benefit of that would be that we remove one step in the amount of migration that implementing apps need to do (changing the junit4 packagename).

Reply:
I think the junit 4 and 5 classes should be separate from each other and from the common classes. That is why I structured it this way, and implemented tests to enforce it. I left the deprecated classes so as to not force changes in depending libs.

Reply:
Sure. The concern was about existing apps that already import classes using the old package name that does not contain`.junit4`. For them, the migration path once they upgrade this lib would be:
1. Remove compilation warnings by updating imports from the old package name to the new package name (which includes `.junit4`)
2. Later, once everything for junit5 is ready: Upgrade all tests to use junit5.
The suggestion above was to remove the need for step 1, and only propose step 2 for the apps.

It's just a suggestion of how to minimize changes in implementing apps. If you feel that the changes in this MR is the best way to do the transition then lets do that.

Reply:
Maybe we should hold off merging this, or at least making a release, until we have more JUnit 5 tools ready so that the teams that want to upgrade can do it in one step